src/test/java/bio/terra/service/snapshotbuilder/SnapshotRequestDaoTest.java

```diff
@@ -0,0 +1,205 @@
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThrows;
+
+import bio.terra.common.EmbeddedDatabaseTest;
+import bio.terra.common.TestUtils;
+import bio.terra.common.category.Unit;
+import bio.terra.common.exception.NotFoundException;
+import bio.terra.common.fixtures.DaoOperations;
+import bio.terra.common.fixtures.ProfileFixtures;
+import bio.terra.common.fixtures.ResourceFixtures;
+import bio.terra.model.BillingProfileModel;
+import bio.terra.model.BillingProfileRequestModel;
+import bio.terra.model.SnapshotAccessRequest;
+import bio.terra.model.SnapshotAccessRequestResponse;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.profile.ProfileDao;
+import bio.terra.service.resourcemanagement.google.GoogleProjectResource;
+import bio.terra.service.resourcemanagement.google.GoogleResourceDao;
+import java.io.IOException;
+import java.util.UUID;
+import org.junit.Assert;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ActiveProfiles({"google", "unittest"})
+@Tag(Unit.TAG)
+@EmbeddedDatabaseTest
+class SnapshotRequestDaoTest {
+
+  @Autowired private DaoOperations daoOperations;
+  @Autowired private ProfileDao profileDao;
+  @Autowired private GoogleResourceDao resourceDao;
+  @Autowired private SnapshotRequestDao snapshotRequestDao;
+
+  private BillingProfileModel billingProfile;
+  private GoogleProjectResource projectResource;
+  private Dataset dataset;
+  private SnapshotAccessRequest snapshotAccessRequest;
+
+  private static final String EMAIL = "user@gmail.com";
+
+  @BeforeEach
+  void setUp() throws IOException {
+    BillingProfileRequestModel profileRequest = ProfileFixtures.randomBillingProfileRequest();
+    billingProfile = profileDao.createBillingProfile(profileRequest, "testUser");
+
+    projectResource = ResourceFixtures.randomProjectResource(billingProfile);
+    UUID projectId = resourceDao.createProject(projectResource);
+    projectResource.id(projectId);
+
+    dataset = daoOperations.createMinimalDataset(billingProfile.getId(), projectId);
+
+    snapshotAccessRequest = SnapshotBuilderTestData.ACCESS_REQUEST;
+  }
+
+  private SnapshotAccessRequestResponse create() {
+    return snapshotRequestDao.create(dataset.getId(), snapshotAccessRequest, EMAIL);
+  }
+
+  private void verifyResponseContents(SnapshotAccessRequestResponse response1) {
```

Comment:
Not sure why this is `response1`, it doesn't seem to conflict with another name in scope.
```suggestion
  private void verifyResponseContents(SnapshotAccessRequestResponse response) {
```